python/core/auto_generated/qgslegendrenderer.sip.in

```diff
@@ -28,7 +28,7 @@ All spacing and sizes are in millimeters.
 %End
   public:
 
-    QgsLegendRenderer( QgsLayerTreeModel *legendModel, const QgsLegendSettings &settings );
+    QgsLegendRenderer( QgsLegendModel *legendModel, const QgsLegendSettings &settings );
```

Comment:
This is an API break -- you can't make this change

Reply:
Hmm I thought so, sadly. Otherwise I could not find a way to skip the label evaluation from the qgslegendrenderer as the call up to the qgslegendmodel is too deep. I can set up some skip check but I'm not too sure if there can be another approach to prevent having to evaluate something three time and simply return the value on the two other calls. :/

Reply:
@nyalldawson would it be fine if I keep a constructor that takes the QgsLayerTreeModel, convert it to a QgsLegendModel ? Or would you suggest to come up with another approach entirely that doesn't rely on having a QgsLegendModel in QgsLegendRenderer for this? I would assume that having an actual QgsLegendModel instead of QgsLayerTReeModel might be desirable..

Reply:
I added a new construction that does the conversion to prevent breakage.

Reply:
@nyalldawson I'm not sure if you were waiting for this to be resolved by the approach is different now and this is not done anymore. The expressions are evaluated once and the output only is reused.